src/server/verification/verification.js

```diff
@@ -19,49 +20,48 @@ class Verifications implements VerificationAPI {
-    this.log.debug('sessionId:', { sessionId })
-    const searchData = Helper.prepareSearchData(verificationData)
-    // log.info('searchData', { searchData })
-    const isDuplicate = await Helper.isDuplicatesExist(searchData, verificationData.enrollmentIdentifier)
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isNotDuplicate')
-      .put(!isDuplicate) // publish to subscribers
 
-    this.log.debug('isDuplicate result:', { user: user.identifier, isDuplicate })
-    if (isDuplicate) return { ok: 1, isDuplicate }
-    const enrollData = Helper.prepareEnrollmentData(verificationData)
-    // log.info('enrollData', { enrollData })
-    const enrollResult: EnrollResult = await Helper.enroll(enrollData)
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isEnrolled')
-      .put(enrollResult.alreadyEnrolled || (enrollResult.enrollmentIdentifier ? true : false)) // publish to subscribers
-    const livenessFailed = (enrollResult && enrollResult.ok === false) || enrollResult.livenessResult === 'undetermined'
-    this.log.debug('liveness result:', { user: user.identifier, livenessFailed })
+    const enrollHandler = async (userId, sessionId, data) => {
+      this.log.debug('Verifying user enrollHandler :', data)
+      const checkFields = ['isDuplicate', 'isLive', 'isEnrolled']
 
-    GunDBPublic.gun
-      .get(sessionId)
-      .get('isLive')
-      .put(!livenessFailed) // publish to subscribers
-    if (livenessFailed) return { ok: 1, livenessPassed: false }
-
-    //this.log.debug('liveness result:', { user: user.identifier, livenessPassed }) // This is left to support future granularity for user better UX experience. Consider using authenticationFacemapIsLowQuality property https://dev.zoomlogin.com/zoomsdk/#/webservice-guide
-    //if (!livenessPassed) return { ok: 1, livenessPassed }
+      for (let field in data) {
+        if (checkFields.indexOf(field) >= 0) {
+          GunDBPublic.gun
+            .get(sessionId)
+            .get(field)
+            .put(data[field])
+        }
+      }
 
-    const isVerified =
-      !isDuplicate && (enrollResult.alreadyEnrolled || (enrollResult.enrollmentIdentifier ? true : false)) // enrollResult.enrollmentIdentifier should return true if there is value in it (and not the value itself) into isVerified.
-    return {
-      ok: 1,
-      isVerified,
-      enrollResult: { ...enrollResult, enrollmentIdentifier: verificationData.enrollmentIdentifier }
+      if (data.ok && data.result) {
+        this.log.debug('Whitelisting new user', user)
+        try {
+          await Promise.all([
+            AdminWallet.whitelistUser(user.gdAddress, user.profilePublickey),
+            storage
+              .updateUser({ identifier: user.loggedInAs, isVerified: true })
+              .then(updatedUser => this.log.debug('updatedUser:', updatedUser))
+          ])
+          await GunDBPublic.gun
+            .get(sessionId)
+            .get('isWhitelisted')
```

Comment:
.put({ isWhitelisted: true })

Reply:
done

Reply:
> 1  the same question - what if 'not ok' ?

I added the condition if data.ok === 0 

> why session is updated to enroll payload before checking data.ok && data.result ?

checking for duplication and for living or not returns {isDuplicate, isLive}
![image](https://user-images.githubusercontent.com/8187382/77072961-2b267280-69f7-11ea-91c0-5b7579f88d40.png)

> GunDBPublic.gun.get(sessionId) ref is used many times so it can be assigned to some constant

**done** use **sessionRef**, done